proxmoxtf/provider/provider.go

```diff
@@ -116,6 +117,11 @@ func providerConfigure(_ context.Context, d *schema.ResourceData) (interface{},
 	sshSocks5Username := utils.GetAnyStringEnv("PROXMOX_VE_SSH_SOCKS5_USERNAME")
 	sshSocks5Password := utils.GetAnyStringEnv("PROXMOX_VE_SSH_SOCKS5_PASSWORD")
 
+	// On Windows, binaries using SSH typically assume a hardcoded name for the ssh-agent socket
+	if runtime.GOOS == "windows" && sshAgentSocket == "" {
+		sshAgentSocket = `\\.\pipe\openssh-ssh-agent`
+	}
+
```

Comment:
I think it would be better to move this block to windows-specific [proxmox/ssh/client_windows.go](https://github.com/bpg/terraform-provider-proxmox/pull/1270/files#diff-166a9f8f1992aad89227877e99fed6ac73dca940ad5bc9cd2081bba9e637639d) inside the `dialSocket` func

Reply:
`dialSocket` will never get an empty address because of this:
https://github.com/bpg/terraform-provider-proxmox/blob/6ae9b581df6e82cfe9bae2bb9174d8cc3b46ea6c/proxmox/ssh/client.go#L534-L537

What if I change this so that it fills in the default socket address on Windows, and returns the error otherwise?

Reply:
Ah, good point, missed that. This check can be moved into platform-specific `dialSocket` as well. So if address is missing we'll supply the default value on Windows or throw an error on other platforms.

Reply:
Made the change, let me know what you think